elephant/test/test_phase_analysis.py

```diff
@@ -370,5 +373,329 @@ def testPhaseLockingValue_raise_Error_if_trial_lengths_are_different(self):
+                self.lfps1_real, self.lfps2_real, self.sf1_real)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_real, atol=1e-14,
+                rtol=1e-12, equal_nan=True)
+        # np.array-input
+        with self.subTest(msg="np.array input"):
+            freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(
+                self.lfps1_real.magnitude, self.lfps2_real.magnitude,
+                self.sf1_real)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_real, atol=1e-14,
+                rtol=1e-12, equal_nan=True)
+        # neo.AnalogSignal-input
+        with self.subTest(msg="neo.AnalogSignal input"):
+            freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(
+                self.lfps1_real_AnalogSignal, self.lfps2_real_AnalogSignal)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_real, atol=1e-14,
+                rtol=1e-12, equal_nan=True)
+
+    def test_WPLI_ground_truth_consistency_artificial_LFP_dataset(self):
+        """
+        Test if the WPLI is consistent with the ground truth generated from
+        artificial LFP-datasets.
+        """
+        # Quantity-input
+        with self.subTest(msg="Quantity input"):
+            freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(
+                self.lfps1_artificial, self.lfps2_artificial,
+                self.sf1_artificial, absolute_value=False)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_artificial,
+                atol=1e-14, rtol=1e-12, equal_nan=True)
+        # np.array-input
+        with self.subTest(msg="np.array input"):
+            freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(
+                self.lfps1_artificial.magnitude,
+                self.lfps2_artificial.magnitude, self.sf1_artificial,
+                absolute_value=False)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_artificial,
+                atol=1e-14, rtol=1e-12, equal_nan=True)
+        # neo.AnalogSignal-input
+        with self.subTest(msg="neo.AnalogSignal input"):
+            freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(
+                self.lfps1_artificial_AnalogSignal,
+                self.lfps2_artificial_AnalogSignal, absolute_value=False)
+            np.testing.assert_allclose(
+                wpli, self.wpli_ground_truth_FieldTrip_artificial,
+                atol=1e-14, rtol=1e-12, equal_nan=True)
+
+    def test_WPLI_comparison_to_multitaper_approaches(self):
+        """
+        Test if WPLI values are equal to those calculated from
+        FieldTrips' ft_connectivity() and MNEs' spectral_connectivity() at
+        frequencies [16, 36, 52, 70, 100]Hz.
+        """
+        for i in range(1, 4):
```

Comment:
The only difference in each sub-test is the variables that are used as parameters for the WPLI function. You can then further improve and take advantage of `subTest`s by using a `dict` to define the parameters for the function and iterate over it. Therefore, you can get the same information as in the branching that you are using, but the code will be less redundant. Not only this reduce errors and improve readability, but also all tests will actually use the same code.

```
tests = {

	'Quantity': {
		'signal_i': self.lfps1_artificial,
		'signal_j': self.lfps2_artificial,
		'sampling_frequency': self.sf1_artificial
	},


	'np.ndarray': {
		'signal_i': self.lfps1_artificial.magnitude,
		'signal_j': self.lfps2_artificial.magnitude,
		'sampling_frequency': self.sf1_artificial
	},


	'neo.AnalogSignal' : {
		'signal_i': self.lfps1_artificial_AnalogSignal,
		'signal_j': self.lfps2_artificial_AnalogSignal
	}


}

for input_type, inputs in tests.items():
	# Compute WPLI using each input type
	freq, wpli = elephant.phase_analysis.weighted_phase_lag_index(**inputs, absolute_value=False)
 
        # mask for supposed wpli=1 frequencies,
        # freq=70Hz with supposed wpli=0 is treated separately,
        # because of noise existence in the artificial dataset
        mask = ((freq == 16) | (freq == 36) | (freq == 52) | (freq == 100))

        # comparing to FieldTrips' ft_conectivity()
        with self.subTest(msg=f"FieldTrip; wpli=1; {input_type} input"):
		...
	...
```

Reply:
commit 131f5abed9c010881d38b502c07875dce39d6b5a adopted this suggestion